base/steer/FairRootManager.h

```diff
@@ -385,6 +396,24 @@ class FairRootManager : public TObject
     /** Iterator for the list of branches used with no-time stamp in time-based session */
     TIterator* fListOfNonTimebasedBranchesIter; //!
 
+//_____________________________________________________________________
+    /**private Members for multi-threading */
+    // methods
+    void  LogMessage(const TString& message);
+    void  FillWithLock();
+    void  FillWithTmpLock();
+    void  FillWithoutLock();
+
+    // global static data members
+    static  Int_t    fgCounter;         // The counter of instances
+    static  Bool_t   fgIsFillLock;      // The if the Fill should be locked
+    static  std::vector<Bool_t>* fgIsFillLocks; // The info per thread if the Fill should be locked
```

Comment:
does this need to be a pointer?

Reply:
This gives us full control on creating/deleting this object; see also the O2 coding guidelines (Global variables initialization) which explicitly forbid static data members of std::vector type.

Reply:
Ok ... but do you need full control? Do you prefer calling "delete" vs let the compiler clean it up?

I could do not find a statement that forbids static std::vector ... (if the O2 coding guidelines apply at all for FairRoot).

Reply:
just to clarify things: The O2 coding guidelines does not apply at all to FairRoot

Reply:
Yes, sure, but the rule was added to the guidelines for a good reason, and this reason is also explained here. See Decision, and then the example in extra details section.
So even if the O2 guidelines do not apply to FairRoot, I would prefer to keep this code as it is.